src/screens/PinScreen.js

```diff
@@ -111,7 +111,7 @@ class PinScreen extends React.Component {
   handleDataMigration = (pin) => {
     const accessData = hathorLib.wallet.getWalletAccessData();
 
-    if (accessData !== null && accessData.xpubkey === undefined) {
+    if (accessData !== null && accessData.xpubkey === undefined && accessData.mainKey) {
```

Comment:
Why do we need this new criteria?

Reply:
This is an old code for a data migration that we did more than a year ago (I thought about removing this code to be honest) where we didn't have the xpubkey saved on storage but now we do.

The wallet service facade does not store the `mainKey` data on storage like the old one, that's why I had to add this new condition.

Reply:
Ah! Ok. Thanks!